### DISCUSSION TITLE
Inconsistent results from Create New Project REST API endpoint

### DISCUSSION THREAD

**Discussion by spenpo** [vercel/community#3724] [Bug Report]
### Page to Investigate

https://vercel.com/docs/rest-api/endpoints#create-a-new-project

### Steps to Reproduce

This post request has been performing inconsistently for me. I have a template repo that is just a landing page configured by env variables. My nextjs app has a serverless function that clones the template with the github api and creates a new vercel project on my account with the vercel rest api and connects it to the newly cloned repo. Most of the time that does not trigger a deployment. Is it intended that creating a new project triggers a new deployment or not?

Here's the rest of the context. The user of my app also has to upload a headshot for the landing page, so after I create the vercel project I'm using the github api again to push a commit with their headshot to the new repo. As expected, that always triggers a deployment, but about 20% of the time that deployment gets stuck in the queue because the post request to create the vercel project actually did trigger a deployment which the headshot deployment is still behind in line. 

The fact that I run into this sporadically in my testing even though the underlying serverless function hasn't really changed lately is quite confusing. I would like to know if this is a bug or if I might be doing something wrong that makes the request unpredictable.

The logic for my serverless function is [here](https://github.com/spope851/spenpo-next/blob/feature/landing/pages/api/deployLanding.ts) 

Thanks for reading!

**Comment by amyegan** [comment]
Hey @spope851. Are you able to share the names of any projects that triggered a deployment at the same time the project was created?

**Reply by spenpo** [reply]
hey @amyegan !
yea I can give an example of both:

- [this one](https://vercel.com/spope851/spenpo-landing-1/deployments) triggered a deployment both when the project was created (Initial commit) and when the headshot commit was pushed up (add headshot)
- [this one](https://vercel.com/spope851/roger-landing/deployments) only deployed from the commit being pushed
